hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestHoodieReadClient.java

```diff
@@ -48,6 +51,35 @@
  */
 public class TestHoodieReadClient extends HoodieClientTestBase {
 
+  private String basePathWithoutScheme = null;
+  private static final String LOCAL_FS_SCHEME = "file://";
+
+  /**
+   * Initializes basePath to use local FS.
+   */
+  protected void initPath() {
+    try {
+      java.nio.file.Path basePath = tempDir.resolve("dataset");
+      java.nio.file.Files.createDirectories(basePath);
+      basePathWithoutScheme = basePath.toString().contains("//") ? basePath.toString().substring(basePath.toString().indexOf("//") + 2) : basePath.toString();
```

Comment:
can we avoid concat fs scheme? please make use of URI. if you do Path#toURI() i think you should get the right scheme